src/ViewRequest.test.js

```diff
@@ -331,37 +331,121 @@ describe('ViewRequest', () => {
           mockedLocation.search = null;
         });
 
-        describe('when current lending request is closed', () => {
-          const openValidRequest = {
-            ...mockedRequestWithDCBUser,
-            status: requestStatuses.FILLED,
-          };
+        describe('when current lending request status starts with "Closed"', () => {
+          it('should not render action menu when request status is "Closed - Filled"', () => {
+            const closedFilledRequest = {
+              ...mockedRequestWithDCBUser,
+              status: requestStatuses.FILLED,
+            };
+
+            const props = {
+              ...alteredProps,
+              resources: {
+                selectedRequest: {
+                  hasLoaded: true,
+                  records: [
+                    {
+                      ...alteredProps.resources.selectedRequest.records,
+                      ...closedFilledRequest,
+                    },
+                  ],
+                },
+              },
+            };
 
-          const props = {
-            ...alteredProps,
-            resources: {
-              selectedRequest: {
-                hasLoaded: true,
-                records: [
-                  {
-                    ...alteredProps.resources.selectedRequest.records,
-                    ...openValidRequest,
-                  },
-                ],
+            render(
```

Comment:
This render is repeated multiple times - let's move it to a function `renderViewRequest` and call it instead

Reply:
Hey @BogdanDenis 
Implemented not only your recommendations and but also more refinement. Please re-review.